e2e/ibft_test.go

```diff
@@ -1,52 +1,68 @@
 package e2e
 
 import (
-	"fmt"
+	"context"
 	"math/big"
+	"os"
 	"testing"
-
-	"github.com/umbracle/go-web3/jsonrpc"
+	"time"
 
 	"github.com/0xPolygon/minimal/crypto"
-	"github.com/0xPolygon/minimal/helper/hex"
+	"github.com/0xPolygon/minimal/e2e/framework"
 	"github.com/0xPolygon/minimal/types"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestIbft_Transfer(t *testing.T) {
-	var privKeyRaw = "0x4b2216c76f1b4c60c44d41986863e7337bc1a317d6a9366adfd8966fe2ac05f6"
-	key, _ := crypto.ParsePrivateKey(hex.MustDecodeHex(privKeyRaw))
-
-	// 0xdf7fd4830f4cc1440b469615e9996e9fde92608f
-	addr := crypto.PubKeyToAddress(&key.PublicKey)
+	signer := &crypto.FrontierSigner{}
+	senderKey, senderAddr := framework.GenerateKeyAndAddr(t)
+	_, receiverAddr := framework.GenerateKeyAndAddr(t)
 
-	clt, err := jsonrpc.NewClient("http://127.0.0.1:10002") /* http://127.0.0.1:8545 */
-	assert.NoError(t, err)
+	dataDir, err := framework.TempDir()
+	if err != nil {
+		t.Fatal(err)
+	}
 
-	eth := clt.Eth()
-	fmt.Println(eth.BlockNumber())
+	ibftManager := framework.NewIBFTServersManager(t, IBFTMinNodes, dataDir, IBFTDirPrefix, func(i int, config *framework.TestServerConfig) {
+		config.Premine(senderAddr, framework.EthToWei(10))
+		config.SetSeal(true)
+	})
+	t.Cleanup(func() {
+		ibftManager.StopServers()
+		if err := os.RemoveAll(dataDir); err != nil {
+			t.Log(err)
+		}
+	})
 
-	signer := crypto.NewEIP155Signer(100)
-	target := types.StringToAddress("0x1010101010101010101010101010101010101010")
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+	ibftManager.StartServers(ctx)
 
+	srv := ibftManager.GetServer(0)
 	for i := 0; i < 3; i++ {
 		txn := &types.Transaction{
-			From:     addr,
-			To:       &target,
+			From:     senderAddr,
+			To:       &receiverAddr,
 			GasPrice: big.NewInt(10000),
 			Gas:      1000000,
-			Value:    big.NewInt(10000),
+			Value:    framework.EthToWei(1),
 			Nonce:    uint64(i),
 		}
-		txn, err = signer.SignTx(txn, key)
+		txn, err = signer.SignTx(txn, senderKey)
 		if err != nil {
-			panic(err)
+			t.Fatal(err)
 		}
 		data := txn.MarshalRLP()
-		fmt.Println(data)
 
-		hash, err := eth.SendRawTransaction(data)
+		hash, err := srv.JSONRPC().Eth().SendRawTransaction(data)
+		assert.NoError(t, err)
+		assert.NotNil(t, hash)
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+		defer cancel()
+		receipt, err := srv.WaitForReceipt(ctx, hash)
+
 		assert.NoError(t, err)
-		fmt.Println(hash)
+		assert.NotNil(t, receipt)
```

Comment:
Maybe add additional check, to make sure it is the same transaction:
```suggestion
		assert.NotNil(t, receipt)
		assert.Equal(t, receipt.TransactionHash, hash)
```

Reply:
I added this assertion